frontend/src/components/EditableText.jsx

```diff
@@ -79,18 +82,103 @@ function EditableText({id, text, rubric, isPartOf, links, fragment, setFragment,
       .catch(console.error);
   };
 
+  const imageRegex = /!\[([^\]]*)\]\(([^)]+)\)/g;
+  let images = [];
+  let Text = text || '';
+  let match;
+  while ((match = imageRegex.exec(Text)) !== null) {
+    images.push({ alt: match[1], src: match[2] });
+  }
+  Text = Text.replace(imageRegex, '');
+
+  const confirmDelete = () => {
+    const { src, alt, internal, name } = deleteTarget;
+    const esc = s => s.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    const mdRx = new RegExp(`!\\[${esc(alt)}\\]\\(${esc(src)}\\)`, 'g');
+    const clean = t => (t || '').replace(mdRx, '').replace(/\n{2,}/g, '\n\n').trim();
+    if (internal) {
+      backend.deleteAttachment(id, name, res => {
+        if (!res.ok) return alert('Error deleting attachment.');
```

Comment:
Don't show notifications to users if you don't provide solutions for them to fix the problem.